File: Data Structures & Algorithms/majority-element/submission-7.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        ans = 0
        bit = [0] * 32
        for num in nums:
            for i in range(32):
                bit[i] += ((num>>i) & 1)
        n = len(nums)
        for i in range(32):
            if bit[i] > n//2:
                if i==31:
                    ans -= (1<<i)
                else:
                    ans |= (1<<i)
        return ans

File: Data Structures & Algorithms/sliding-window-maximum/submission-7.java
class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
        ArrayDeque<Integer> dq = new ArrayDeque<>();
        int n = nums.length;
        int[] ans = new int[n-k+1];
        int idx = 0;
        for(int r=0; r<n; r++) {
            int l = r-k+1;
            while(!dq.isEmpty() && nums[dq.peekFirst()]<nums[r]) {
                dq.pollFirst();
            }
            dq.offerFirst(r);
            if(dq.peekLast()<l)
                dq.pollLast();
            if(r>=k-1)
                ans[idx++] = nums[dq.peekLast()];
        }
        return ans;
    }
}
